lib/publiccloud/instance.pm

```diff
@@ -68,9 +70,13 @@ sub run_ssh_command {
     delete($args{no_quote});
     delete($args{ssh_opts});
     delete($args{username});
+    delete($args{rc_only});
     if ($args{timeout} == 0) {
         # Run the command and don't wait for it - no output nor returncode here
         script_run($ssh_cmd, %args);
+    } elsif ($rc_only) {
+        # Run the command and return only the returncode here
+        return script_run($ssh_cmd, %args);
```

Comment:
I would `die()` on timeout.
```suggestion
        my $ret = script_run($ssh_cmd, %args);
        die("Timeout on $ssh_cmd") unless(defined($ret));
        return $ret;
```

Reply:
I can do this yes. In that case I will also have to remove the `defined` test I do in my `wait_until_resources_started` function, not needed anymore.